test/framework/helpers.go

```diff
@@ -94,7 +94,7 @@ func (f *Framework) WaitForPodsReady(ctx context.Context, namespace string, time
 		for _, p := range pl.Items {
 			isRunningAndReady, err := k8sutil.PodRunningAndReady(p)
 			if err != nil {
-				return false, err
+				return false, nil
```

Comment:
so from what I can see, it looks like we started to hit a race condition here with Deletions for thanos ruler in the latest version of K8s 

Essentially, Pods were being Completed before this code was executed so it would always exit with an error.

Much of the use of this function checks for 0 replicas so it might be best to split it into something like WaitForListEmpty and check that the ListOptions has no Pods to support that use case.

Reply:
Agreed. We could also use foreground deletion in the Delete*AndWaitUntilGone() functions and wait that the top-level resource is gone rather than checking (indirectly) that the pods are missing.